homeassistant/components/vesync/number.py

```diff
@@ -0,0 +1,114 @@
+"""Support for number settings on VeSync devices."""
+import logging
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .common import VeSyncDevice
+from .const import DOMAIN, VS_DISCOVERY, VS_DISPATCHERS, VS_NUMBERS
+
+_LOGGER = logging.getLogger(__name__)
+
+DEV_TYPE_TO_HA = {
+    "Classic300S": "humidifier",
+}
+
+MAX_MIST_LEVEL = 9
+MIN_MIST_LEVEL = 1
+MIST_LEVEL_STEP = 1
+
+MAX_NIGHT_LIGHT_BRIGHTNESS = 100
+MIN_NIGHT_LIGHT_BRIGHTNESS = 0
+NIGHT_LIGHT_BRIGHTNESS_STEP = 1
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up numbers."""
+
+    async def async_discover(devices):
+        """Add new devices to platform."""
+        _async_setup_entities(devices, async_add_entities)
+
+    disp = async_dispatcher_connect(
+        hass, VS_DISCOVERY.format(VS_NUMBERS), async_discover
+    )
+    hass.data[DOMAIN][VS_DISPATCHERS].append(disp)
+
+    _async_setup_entities(hass.data[DOMAIN][VS_NUMBERS], async_add_entities)
+
+
+@callback
+def _async_setup_entities(devices, async_add_entities):
+    """Check if device is online and add entity."""
+    entities = []
+    for dev in devices:
+        if DEV_TYPE_TO_HA.get(dev.device_type) == "humidifier":
+            entities.append(VeSyncHumidifierMistLevelHA(dev))
+            if dev.night_light:
+                entities.append(VeSyncHumidifierNightLightHA(dev))
+        else:
+            _LOGGER.debug(
+                "%s - Unknown device type - %s", dev.device_name, dev.device_type
+            )
+            continue
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class VeSyncHumidifierNightLightHA(VeSyncDevice, NumberEntity):
+    """Representation of the night light on a VeSync humidifier.
+
+    This is a number entity instead of a light entity because it does not
+    support an on/off toggle.
+    """
+
+    _attr_max_value = MAX_NIGHT_LIGHT_BRIGHTNESS
+    _attr_min_value = MIN_NIGHT_LIGHT_BRIGHTNESS
+    _attr_step = NIGHT_LIGHT_BRIGHTNESS_STEP
+
+    @property
+    def unique_id(self):
+        """Return the ID of this device."""
+        return f"{super().unique_id}.night_light"
+
+    @property
```

Comment:
Mark entity as `config` (see docs under `entity_category` [here](https://developers.home-assistant.io/docs/core/entity#generic-properties))

```suggestion
    @property
    def entity_category(self):
        """Return the config entity category."""
        return EntityCategory.CONFIG
    
    @property
```